src/adagrams.js

```diff
@@ -1,15 +1,146 @@
+//Get an array of letters, where every letter repeats depending on its frequency
+
+// import { splice } from "core-js/core/array";
+
+// can be done by hand (actually it would take less time)
+export const buildArrayLetters = function (letterDistribution) {
+  const lettersList = [];
+  for (const letter in letterDistribution) {
+    while (letterDistribution[letter] > 0) {
+      lettersList.push(letter);
+      letterDistribution[letter] = letterDistribution[letter] - 1;
+    }
+  }
+  return lettersList;
+};
+
+// get ten random elements from a given array and return an array with
+// those ten elements
+export const getTenRandom = function (listOfLetters) {
+  const randomLetters = [];
+  for (let i = 0; i < 10; i++) {
+    let randomIndex = [Math.floor(Math.random() * listOfLetters.length)];
+    randomLetters.push(listOfLetters[randomIndex]);
+    listOfLetters.splice(randomIndex, 1);
+  }
```

Comment:
Nice C-Style `for` loop! One small note - the `randomIndex` can be made a `const`.